src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -32,95 +32,177 @@
+* with changes to the Deceased and Do Not Contact fields for the Contact object.
 */
 public class CON_DoNotContact_TDTM extends TDTM_Runnable {
 
     /*******************************************************************************************************
-    * @description Handles deceased and do not contact fields.
-    * @param listNew the list of Accounts from trigger new.
-    * @param listOld the list of Accounts from trigger old.
-    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
-    * @param objResult the describe for Accounts
+    * @description Synchronizes Household naming exclusions and communication preferences 
+    * with changes to the Deceased and Do Not Contact fields for the Contact object.
+    * @param newList the list of Contacts from trigger new
+    * @param oldList the list of Contacts from trigger old
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.)
+    * @param objResult the describe for Contacts
     * @return dmlWrapper.
     ********************************************************************************************************/
-    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
+    public override DmlWrapper run(List<SObject> newList, List<SObject> oldList,
         TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
 
-        for (integer i=0; i<newlist.size();i++) {
+        List<Contact> newContactsList = (List<Contact>)newList;
 
-            Contact con = (Contact)newlist[i];
-    
-            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-                if (con.Deceased__c)
-                    markContactDeceased(con);
-                else if (con.Do_Not_Contact__c)
-                    markContactDoNotContact(con);
-            }
-
-            if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
-                Contact oldCon = (Contact)oldlist[i];
+        if (this.isBeforeInsert(triggerAction)) {
+            this.processBeforeInsert(newContactsList);
+        }
 
-                //deceased and do not contact: updates related fields for both checking and unchecking the box.
-                //prefers deceased as the deceased already marks the Do Not Contact fields, in case the user changes both values.
-                if (con.Deceased__c != oldCon.Deceased__c)
-                    markContactDeceased(con);
-                else if (con.Do_Not_Contact__c != oldCon.Do_Not_Contact__c)
-                    markContactDoNotContact(con);
-            }
+        if (this.isBeforeUpdate(triggerAction)) {
+            Map<Id, Contact> oldUpdatedContactsById = new Map<Id, Contact>((List<Contact>)oldlist);
+            this.processBeforeUpdate(newContactsList, oldUpdatedContactsById);
         }
 
         return new DmlWrapper();
-	}
+    }
+    
+    /**
+     * @description Determine if trigger fired is the result of a BeforeInsert action. 
```

Comment:
Sorry about being picky but it looks like other description has different spacing for * and /. Shall we make them consistent. Maybe we can align the / with the first stars.